### ISSUE TITLE
Miner commits more than 1 bid per block and slashes probability to win

### ISSUE THREAD

**Issue by MaksimalistT** [blockstack/stacks-blockchain#2718]
**Describe the bug**
Miner commits more than 1 bid per block and slashes probability to win

**Steps To Reproduce**
bug appears right after the start
here is testnet results for miner: ST39F7SA0A
<img width="746" alt="Screen Shot 2021-06-17 at 6 21 32 PM" src="https://user-images.githubusercontent.com/77570471/122426367-de826c00-cf98-11eb-9ff0-c8961ab862f3.png">

**Environment (please complete the following information):**
- OS: Amazon Linux
- Rust version: rustc 1.49.0 (e1884a8e3 2020-12-29)
 - Stacks-node 2.0.11.1.0

**Comment by jcnelson** [comment]
Can you post both Bitcoin testnet transactions (or even just the txids)?  It's possible that the miner's transaction was just late -- i.e. it was sent at height X, but mined in Bitcoin block X+2 or later.

**Comment by jcnelson** [comment]
Like, this might just be a dup of #2706

**Comment by MaksimalistT** [comment]
i don't know which transaction ids where sent more than once for a block
you can dig deeper by analyzing transactions for the wallet https://tbtc.bitaps.com/mzkNNVhtTF9AJfJTawYmbsL4WfoPngk7qv

**Comment by MaksimalistT** [comment]
> Like, this might just be a dup of #2706

can this affect leader election probabilities?

**Comment by MaksimalistT** [comment]
same happens for node v2.0.11.0.0
<img width="728" alt="Screen Shot 2021-06-17 at 9 36 46 PM" src="https://user-images.githubusercontent.com/77570471/122454604-41353100-cfb4-11eb-814e-9fff5025b6ad.png">

**Comment by MaksimalistT** [comment]
@jcnelson issue was reproduced twice right after restart for v2.0.11.0.0 and v2.0.11.1.0, it looks like a systematic problem
and could you please tell if this can affect leader election probabilities?

**Comment by jcnelson** [comment]
I'm still pretty sure this is a dup of #2706.  The Stacks miner is sending transactions for the targeted Bitcoin block height, but the Bitcoin transactions themselves can get bumped into later blocks by Bitcoin miners for a whole host of reasons (including the tx fee being too low).  If this happens, then yes -- your odds of winning will decrease, since it looks like you didn't mine for a block.  If you can provide me with txids, I can take a closer look at the OP_RETURNs.

**Comment by MaksimalistT** [comment]
> I'm still pretty sure this is a dup of #2706. The Stacks miner is sending transactions for the targeted Bitcoin block height, but the Bitcoin transactions themselves can get bumped into later blocks by Bitcoin miners for a whole host of reasons (including the tx fee being too low). If this happens, then yes -- your odds of winning will decrease, since it looks like you didn't mine for a block. If you can provide me with txids, I can take a closer look at the OP_RETURNs.

please note that bug was reproduced on a testnet

if transaction was bumped to the next block isn't it resulted in a missed stack block for the miner, which will be rendered in a stacks-dump report?

tx ids:
5e3c79c981497761b6335422c7d37d7ef0d0c6b58853ab61b00146da1b3dc2b4
c9314998a7069e46168427ee83bd436a26b5346f1401244b08ebc40f09f91fcf

**Comment by MaksimalistT** [comment]
@jcnelson take a look at this weird stacks and btc block sequence

<img width="746" alt="Screen Shot 2021-06-17 at 6 21 32 PM copy" src="https://user-images.githubusercontent.com/77570471/122606261-5035e500-d081-11eb-8c95-6e0f5290fc4f.png">

**Comment by jcnelson** [comment]
> please note that bug was reproduced on a testnet

Transactions being bumped by Bitcoin miners isn't a bug.  If this is what happened, then there's quite literally nothing we can do about it.

Thank you for providing the txids.  They were definitely mined for two different Stacks blocks, building off two different parents.  5e3c79c981497761b6335422c7d37d7ef0d0c6b58853ab61b00146da1b3dc2b4 was trying to build off of a Stacks block mined in Bitcoin testnet block 2005177, whereas c9314998a7069e46168427ee83bd436a26b5346f1401244b08ebc40f09f91fcf was trying to build off of a Stacks block mined in 2005190.  Considering that both these transactions got confirmed in Bitcoin testnet block 2005191, it appears that c9314998a7069e46168427ee83bd436a26b5346f1401244b08ebc40f09f91fcf was on-time, but 5e3c79c981497761b6335422c7d37d7ef0d0c6b58853ab61b00146da1b3dc2b4 was either mined really late (maybe there was a testnet block storm or something that prevented miners from picking it up?), or was sent by your Stacks node before it had fully finished synchronizing.  Do you know if this was the case?  If so, then you can correct for this by using the `--mine-at-height` command-line option, to force the node to only begin mining after a particular Stacks blockchain height has been downloaded.

Do you have debug logs for your node for when these two transactions were generated?

**Comment by MaksimalistT** [comment]
Hi, @jcnelson, @diwakergupta  i've done series of tests and it happens every time 
there are more than one transaction for one block right after restart
and it regularly shows see more than 1 pending transaction in explorer which not always result in having more than 1 transaction for 1 block 
i am sure it requires closer look from dev and qa team

**Comment by MaksimalistT** [comment]
it reproduces again on testnet for Release 2.0.11.2.0-rc1
2 pending transactions for 1 upcoming block

<img width="1342" alt="Screen Shot 2021-07-14 at 11 23 06 AM" src="https://user-images.githubusercontent.com/77570471/125589432-43dbd16d-2d1f-4315-99d4-20e63c91e56d.png">

**Comment by diwakergupta** [comment]
> it reproduces again on testnet for Release 2.0.11.2.0-rc1
> 2 pending transactions for 1 upcoming block
> 
> **Steps To Reproduce**
> it appears when rbf is executed by node, every call produces new pending transaction

@kantai could you take a look and confirm if it's a dup of #2706 ?

**Comment by kantai** [comment]
I don't think this is a dupe of #2706, but I also am not sure there's any bugged or negative behavior being described here.

The behavior of the stacks-node miner is to RBF when new Stacks information is available since the initial commit was made: whether that's a microblock that can be confirmed, or new transactions that can be included in the miner's anchor block. This seems to be what the linked explorer is displaying. However, "multiple bids" are _not_ being submitted. _Only one_ of the RBF transactions is ever going to be mined in a Bitcoin block. This is the correct behavior. Looking at the history of the discussed wallet:

https://tbtc.bitaps.com/mzkNNVhtTF9AJfJTawYmbsL4WfoPngk7qv

We can see that only one transaction is getting mined per block (the expected behavior).

**Comment by MaksimalistT** [comment]
Hi, @kantai, thanks for investigation
I thought that RBF goal is to increase tx fee and make sure it's moved to the next block and not lost when missing current block.

**Comment by jcnelson** [comment]
This might have been caused by stale `ProcessTenure` requests building up in the relayer channel, which is fixed in the latest release candidate (2.05.0.1.0-rc4).  I'm going to go ahead and close this issue for now; please re-open if this happens again.
